src/components/DateInput/dateInput.test.js

```diff
@@ -3,155 +3,106 @@ import { render, screen, fireEvent } from '@testing-library/react'
-    it('fails incomplete or invalid date strings', () => {
-      expect(isValidUserDate('4', 'MDY')).toBe(false);
-      expect(isValidUserDate('4/20', 'MDY')).toBe(false);
-      expect(isValidUserDate('4/20/10', 'MDY')).toBe(false);
-      expect(isValidUserDate('4/20/10', 'MDY')).toBe(false);
-      expect(isValidUserDate('4/20/201', 'MDY')).toBe(false);
-      expect(isValidUserDate('201/4/20', 'YMD')).toBe(false);
-      expect(isValidUserDate('20/4/201', 'DMY')).toBe(false);
-    });
+  afterEach(() => {
+    dateChangeFn = null;
+    inputChangeFn = null;
   });
 
-  describe('input and picker interation', () => {
-    let dateChangeFn;
-    let inputChangeFn;
-
-    beforeEach(() => {
-      dateChangeFn = jest.fn();
-      inputChangeFn = jest.fn();
-      render(<DateInput onDateChange={dateChangeFn} onInputChange={inputChangeFn} />);
-      fireEvent.click(screen.getByLabelText('Date Input'));
-    });
+  it('calls onInputChange when the input value changes', async () => {
+    expect(inputChangeFn).not.toHaveBeenCalled();
+    await userEvent.type(screen.getByRole('textbox'), '4')
+    expect(inputChangeFn).toHaveBeenCalled();
+  });
 
-    afterEach(() => {
-      dateChangeFn = null;
-      inputChangeFn = null;
-    });
+  it('calls onDateChange when user types a VALID freeform date', async () => {
+    expect(dateChangeFn).not.toHaveBeenCalled();
+    await userEvent.type(screen.getByRole('textbox'), '4/20/2020')
+    expect(dateChangeFn).toHaveBeenCalled();
+  });
 
-    it('calls onInputChange when the input value changes', async () => {
-      expect(inputChangeFn).not.toHaveBeenCalled();
-      await userEvent.type(screen.getByRole('textbox'), '4')
-      expect(inputChangeFn).toHaveBeenCalled();
-    });
+  it('does not call onDateChange when user types an INCOMPLETE freeform date', async () => {
+    expect(dateChangeFn).not.toHaveBeenCalled();
+    await userEvent.type(screen.getByRole('textbox'), '2/3')
+    expect(dateChangeFn).not.toHaveBeenCalled();
+  });
 
-    it('calls onDateChange when user types a VALID freeform date', async () => {
-      expect(dateChangeFn).not.toHaveBeenCalled();
-      await userEvent.type(screen.getByRole('textbox'), '4/20/2020')
-      expect(dateChangeFn).toHaveBeenCalled();
-    });
+  it('does not call onDateChange when user types an INVALID freeform date', async () => {
+    expect(dateChangeFn).not.toHaveBeenCalled();
+    await userEvent.type(screen.getByRole('textbox'), '2/30/2020')
+    expect(dateChangeFn).not.toHaveBeenCalled();
+  });
 
-    it('does not call onDateChange when user types an INCOMPLETE freeform date', async () => {
-      expect(dateChangeFn).not.toHaveBeenCalled();
-      await userEvent.type(screen.getByRole('textbox'), '2/3')
-      expect(dateChangeFn).not.toHaveBeenCalled();
-    });
+  it('calls onDateChange when user selects a day in the picker', () => {
+    expect(dateChangeFn).not.toHaveBeenCalled();
+    fireEvent.click(screen.getByText('7'));
```

Comment:
```suggestion
     fireEvent.click(screen.getByRole('gridcell', { name: 'Tue Jul 07 2020' }));
```

Reply:
I should have updated these in my tests on master. You may have to rebase / merge but you shouldn't need to use `getByText` in any of the tests in this file.

Reply:
did a fresh merge from master; should be resolved now